### ISSUE TITLE
ColorVariable too strict? Using colors in a mix function.

### ISSUE THREAD

**Issue by evoactivity** [brigade/scss-lint#479]
I'm trying to do this

```scss
$form-button-bg: mix($color-variable, #000);
```
But I get a warning 
`[W] ColorVariable:  Color literals like #000 should only be used in variable declarations; they should be referred to via variable everywhere else.`

This seems a little too strict, I like enforcing general color usage to be done with variables, but I don't want to create a generic $black variable just to use #000 to use in sass color functions, especially when they are assigning to a variable.

**Comment by chris-dura** [comment]
> but I don't want to create a generic $black variable just to use #000 to use in sass color functions

Why not? I agree with you that this seems a little silly:
```
$black: #000;
```
But, it's still typically a better practice than using raw hex values...

But, that being said, it actually negates a good benefit of using variables, which is the opportunity to apply some semantic meaning to these color values. Think about how you're actually trying to use the color "black". For example, imagine it's going to be the default/primary text color for your site. What if, for some reason, you want to change the look and feel so that now the text color is just a really, _really_ dark grey? You now have to go replace all the instances of `#000` with `#333`. And having `$black: #333;` is an obvious smell.

So, instead you should consider an approach where you actually attach some semantics to the variable, so even if you change the color of your site from black to green, you could just change it in one place, and the actual CSS rules are still meaningful... just as a for instance.
```
$primary-color: #000;
$inverse-color: #fff;

$form-button-bg: mix($color-variable, $primary-color);
```

**Comment by sds** [comment]
I think @chris-dura captured a lot of what I would normally contribute to this discussion.

One thing I will acknowledge is that perhaps we could relax the check **if you are assigning the result to a variable** (though that starts to sound like we're making the linter itself less useful at capturing duplicate uses of a hex color). It's entirely possible that you have some clever `mix` function which you use to generate all of your core colors. However, ideally you would isolate those to a single file and just disable the `ColorVariable` linter in that file. 

Does that sound reasonable?

**Comment by evoactivity** [comment]
That make sense. I think my problem comes with how I use mix, I often use it in place of lighten and darken as I think it produces a more aesthetically pleasing result, so I only use it with black and white as the secondary color so there really isn't any semantic meaning for that colour. I suppose I could write my own lighten and darken functions that use mix instead.
